include/sys/zfs_context.h

```diff
@@ -65,6 +65,7 @@
 #include <sys/procfs_list.h>
 #include <linux/dcache_compat.h>
 #include <linux/utsname_compat.h>
+#include <linux/mod_compat.h>
```

Comment:
i'd like avoid platform (like 'linux/' and other) specific definitions in shared files - probably create wrapper and use per platform

Reply:
These are all under `#ifdef __linux__` on FreeBSD - although we can move `zfs_context.h` under `os/` we've gotten the linux includes out of the common sources so far. That seems like it should be a separate change as I have largely the headers unchanged. I have tried to keep this commit to be mostly a move.